### PR TITLE
[SPARK-23852][SQL] Upgrade to Parquet 1.8.3

### PR DESCRIPTION
**Pull request by henryr**
## What changes were proposed in this pull request?

Upgrade Parquet dependency to 1.8.3 to avoid PARQUET-1217

## How was this patch tested?

Ran the included new test case.

### CODE REVIEW

**Comment by rdblue** [comment]
@henryr, why not backport the test case in this commit? I don't think it makes sense to separate the two because that test verifies this commit.

**Comment by rdblue** [comment]
+1 when tests are passing.

**Comment by henryr** [comment]
Sounds good, done.

**Comment by gatorsmile** [comment]
Apache Parquet 1.8.3 release only contains https://github.com/apache/parquet-mr/pull/465 and https://github.com/apache/parquet-mr/pull/468, right?

**Comment by vanzin** [comment]
LGTM pending tests.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala]: Since this test case assumes `spark.sql.parquet.filterPushdown=true`, let's use the followings.
```scala
withSQLConf(SQLConf.PARQUET_FILTER_PUSHDOWN_ENABLED.key -> "true",
```

**Comment by gatorsmile** [comment]
cc @liancheng @michal-databricks @cloud-fan Please double check and confirm the risk of these two Parquet PRs is low.

**Comment by vanzin** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala]: +1

**Review by HyukjinKwon — APPROVED** [review]
LGTM too

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala
Comment (vanzin) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala]: That should be done in master (and backported to 2.3 if desired).

**Comment by vanzin** [comment]
Any remaining feedback here? Otherwise I'd like to get this in before soon-ish.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala
Comment (henryr) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala]: PR for master is https://github.com/apache/spark/pull/21323. My guess is there's no reason to block this backport and 2.3.1 by waiting for it to land, but happy to do whatever.

**Comment by vanzin** [comment]
@henryr could you update the PR description (part about the test backport)? Thx

**Comment by henryr** [comment]
Done.

**Comment by vanzin** [comment]
Merging to 2.3. In the unlikely event of issues, we can address them later.

**Comment by vanzin** [comment]
Also, please close the PR manually (github doesn't do that for branches).

**Pull request closed without merging**
